scripts/files/fs_s3.py

```diff
@@ -0,0 +1,101 @@
+from typing import TYPE_CHECKING
+
+import boto3
+import botocore
+from linz_logger import get_log
+
+from scripts.aws.aws_helper import get_credentials, parse_path
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.service_resource import S3ServiceResource
+else:
+    S3ServiceResource = object
+
+
+def get_auth_s3_session(bucket_name: str) -> S3ServiceResource:
+    """Return a boto3 S3 Resource with the AWS credentials for a bucket.
+
+    Args:
+        bucket_name (str): The name of the bucket.
+
+    Returns:
+        S3ServiceResource: The boto3 S3 Resource.
+    """
+    credentials = get_credentials(bucket_name)
```

Comment:
you could try reading/writing the location before creating credentials as the default creds may have enough power to read or write.

Reply:
That's what it's trying to do in the `read` or `write` methods actually